File: MyKosinka/src/mykosinka/Stopka.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package mykosinka;

import java.util.ArrayList;
import java.util.List;
/*
 *
 * @author sakazi
 */
public class Stopka {
    
    List <Card> stopka=new ArrayList<>();
    public static int numStopok=7;//количество стопок
    
    
    private void StopkaStopka(){
        
    }
    
    private void StopkaDom(){
        
    }
    
    public void output(){
        int size = stopka.size();
        for(int i=0;i<size;++i){
            stopka.get(i).output();
            System.out.printf(" ");
        }
        System.out.println();
     }
    public void openLastCard(){
        if(stopka.isEmpty()) return;
        stopka.get(stopka.size()-1).setSostoyanie(1);
    }
    
    public Card getLastCard(){
        if(stopka.isEmpty()) return null;
        return stopka.get(stopka.size()-1);
    }
    
    public Card getFirstCard(){
        if(stopka.isEmpty()) return null;
        return stopka.get(0);
    }
    
    public void closeCards(){
        int size=stopka.size();
        for(int i=0;i<size;++i) 
            stopka.get(i).setSostoyanie(0);
    }
}

File: berkunsky-kosynka/stopka.cpp
#include "stopka.h"
#include <iostream>

int Stopka::numStopok=7;

Stopka::Stopka():cards(0)
{
    cards = new std::vector<Card*>;
}

Stopka::Stopka(const Stopka& other)
{

}

Stopka::~Stopka()
{

}

Stopka& Stopka::operator=(const Stopka& other)
{

}

bool Stopka::operator==(const Stopka& other)
{

}

/*
void Stopka::output()
{
    int size = cards->size();
    for(int i=0; i<size; ++i) {
        cards->at(i)->output();
        std::cout<<" ";
    }
    std::cout<<"\n";
    std::cout.flush();
}
*/

std::ostream & operator<<(std::ostream &os, Stopka &s)
{
    int size = s.cards->size();
    for(int i=0; i<size; ++i) 
    {
        os<<*s.cards->at(i);//->output();
        os<<" ";
    }
    os<<"\n";
    os.flush();
}

void Stopka::openLastCard()
{
    if(cards->empty()) return;
    cards->at(cards->size()-1)->setSostoyanie(1);
}

Card* Stopka::getLastCard()
{
    if(cards->empty()) return NULL;
    return cards->at(cards->size()-1);
}

Card* Stopka::getFirstCard()
{
    if(cards->empty()) return NULL;
    return cards->at(0);
}

void Stopka::closeCards()
{
    int size=cards->size();
    for(int i=0; i<size; ++i)
        cards->at(i)->setSostoyanie(0);
}

void Stopka::StopkaStopka()
{

}

void Stopka::StopkaDom() {

}
